tools/backups2datalad/zarr.py

```diff
@@ -89,9 +106,6 @@ async def run(self) -> None:
             async for entry in ait:
                 log.info("Zarr %s: %s: Syncing", self.zarr_id, entry)
                 self.extant_paths.add(str(entry))
```

Comment:
re my comments about "fetching only what has changed since the last time" ([ref last post](https://github.com/dandi/dandisets/pull/158#issuecomment-1132926402) in that "thread") -- can't we are this point , if we know prior run `.last_timestamp` just `continue` here if the `"LastModified"` of this entry is not after the `.last_timestamp`? 
That would avoid needing for doing all the IO (possibly on thousands of local files in deep directories) below until it is needed for a new(er) file.  

re "prior run `.last_timestamp`" I think we use it for a commit datetime, but I think it is better to not use it for this purpose and better have it explicitly stored in e.g. `.dandi/s3sync.json`, with `prefix` and `last_modified` . If `prefix` mismatch -- error (exception) out since something must have changed whenever it should have not.